Palindrome.java

```diff
@@ -0,0 +1,132 @@
+package WEEK1;
+
+import static org.testng.Assert.assertEquals;
+
+//import java.util.Scanner;
+
+import org.junit.Test;
+
+public class Palindrome {
+	static int length = 0;
+	
+	public static boolean is_palindrome(Node first, Node second){
+		Node temp_f = first;
+		Node temp_s = second;
+		
+		while(temp_f!=null && temp_f.next!=second){
+			if(temp_f.data == temp_s.data){
+				temp_f = temp_f.next;
+				temp_s = temp_s.next;
+			}
+			else{
+				return false;
+			}
+		}
+		return true;
+	}
+	
+	public static Node reverse(Node start){
+		Node current = start;
+		Node prev = null;
+		Node next;
+		while(current!=null){
+			next = current.next;
+			current.next = prev;
+			prev = current;
+			current = next;
+		}
+		
+		start = prev;
+		return start;
+	}
+	
+	public static boolean palindrome(Node start){
+		if(start==null){
+			return true;
+		}
+		if(start.next==null){
+			return false;
+		}
+		Node last = start;
+		Node temp = start;
+		int n = (length/2);
```

Comment:
For this problem, you can (and should) solve it without being given this information.  You should also avoid using static variables unless absolutely necessary as they can be changed at any time and make programs difficult to reason about.

https://stackoverflow.com/questions/7026507/why-are-static-variables-considered-evil

Reply:
https://stackoverflow.com/questions/16446355/how-to-find-a-middle-element-in-a-link-list-without-traversing-the-entire-list